apps/new/widget/page/projects/CatalogImport.jsx

```diff
@@ -0,0 +1,193 @@
+  margin-left: 24px;
+  margin-top: 24px;
+`;
+
+const TagSkeleton = styled.div`
+  background: #eee;
+  border-radius: 4px;
+  height: 34px;
+  width: 110px;
+  margin: 24px;
+`;
+
+const CardSkeleton = () => (
+  <CardSkeletonContainer>
+    <HeaderSkeleton />
+    <TitleSkeleton />
+    <DescriptionSkeleton />
+    <TagSkeleton />
+  </CardSkeletonContainer>
+);
+
+query = fetch(indexer + "/projects");
+if (!query || !query.body) {
+  return "Loading...";
+}
+projects = query.body;
+
+if (!projects) {
+  return "No projects found";
+}
+
+const Container = styled.div`
+  .error-bg {
+    background-color: #4d4d4d;
+    color: white;
+  }
+  max-width: 100%;
+`;
+
+const [filteredProjects, setFilteredProjects] = useState(projects);
+const [searchTerm, setSearch] = useState(null);
+
+const Search = useMemo(() => {
+  return (
+    <Widget
+      src={"${alias_new}/widget/page.projects.SearchBar"}
+      props={{
+        title: sort,
+        numItems: filteredProjects.length,
+        term: searchTerm,
+        itemName: "project",
+        onSearch: (value) => {
+          setSearch(value);
+
+          if (searchTerm === value) return;
+          if (searchTerm === "") return;
+
+          const filtered = {};
+          Object.keys(projects).forEach((projectId) => {
+            if (
+              projects[projectId].profile.name
+                .toLowerCase()
+                .includes(value.toLowerCase())
+            ) {
+              filtered[projectId] = projects[projectId];
+            }
+
+            if (
+              projects[projectId].profile.tagline
+                .toLowerCase()
+                .includes(value.toLowerCase())
+            ) {
+              filtered[projectId] = projects[projectId];
+            }
+
+            const tags = Object.values(projects[projectId].profile.tags);
+            if (
+              tags.some((tag) =>
+                tag.toLowerCase().includes(value.toLowerCase()),
+              )
+            ) {
+              filtered[projectId] = projects[projectId];
+            }
+          });
+          setFilteredProjects(filtered);
+        },
+      }}
+    />
+  );
+}, [projects, searchTerm]);
+
+const ProjectsGrid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, minmax(0, 1fr));
+  gap: 1.5rem;
+  @media screen and (max-width: 768px) {
+    grid-template-columns: repeat(1, minmax(0, 1fr));
+  }
+  max-width: 100%;
+`;
+
+const [selectedProjectId, setSelectedProjectId] = useState(null);
+
+if (selectedProjectId) {
+  return (
+    <Widget
+      src="${alias_new}/widget/page.projects.Editor"
+      loading=""
+      props={{
+        catalogProjectId: selectedProjectId,
+      }}
+    />
+  );
+}
+
+return (
+  <Container className="d-flex flex-column gap-4 p-4">
+    {Search}
+    {Object.keys(filteredProjects).length === 0 && (
+      <div className="error-bg p-3 h6 rounded-3">
+        {searchTerm
+          ? "No projects were found for your search query."
```

Comment:
Add test cases to cover both scenarios

"should show no projects found if non-existing search term provided"
"should show network issue error if no search term and no projects found"